src/util.rs

```diff
@@ -243,3 +253,56 @@ macro_rules! row_col_vec {
         }
     };
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_move_grammar() {
+        // move_grammar(map: &mut HashMap<Coordinate, Grammar>, source: Coordinate, dest: Coordinate)
+        unimplemented!();
+    }
+
+    #[test]
+    fn test_non_zero_u32_tuple() {
+        
+        assert_eq!(non_zero_u32_tuple((1, 2)),(NonZeroU32::new(1).unwrap(), NonZeroU32::new(2).unwrap()));
+        assert_ne!(non_zero_u32_tuple((1, 2)), (NonZeroU32::new(2).unwrap(), NonZeroU32::new(2).unwrap()));
+        // unimplemented!();
+    }
+
+    #[test]
+    fn test_row_col_to_string() {
+        assert_eq!(row_col_to_string((2,2)), "B2");
+        assert_ne!(row_col_to_string((2,2)), "A2");
+        // unimplemented!();
+    }
+
+    #[test]
+    fn test_coord_show() {
+        assert_eq!(coord_show(vec![(1, 1), (1, 1)]).unwrap(), "root-A1");
+        assert_ne!(coord_show(vec![(1, 1), (1, 1)]).unwrap(), "root")
+        // unimplemented!();
+    }
+
+    #[test]
+    fn test_apply_definition_grammar() { // will be tested on Integration
+        unimplemented!();
+    }
+
+    #[test]
+    fn test_resize() { // will be tested on Integration
+        unimplemented!();
+    }
+
+    #[test]
+    fn test_resize_diff() { // will be tested on Integration
+        unimplemented!();
+    }
+
+    #[test]
+    fn test_dom_resize() { // will be tested on Integration
+        unimplemented!();
+    }
+}
```

Comment:
for these tests that will be in integration, don't bother putting them in unit tests at all. so
```suggestion

```